### DISCUSSION TITLE
Unable to detect metamask wallet when other wallet is installed

### DISCUSSION THREAD

**Discussion by johnpaulcas** [family/connectkit#258] [Q&A] [answered]
![Screenshot 2023-07-05 at 2 29 03 PM](https://github.com/family/connectkit/assets/44362950/3a27cc5a-6323-416b-ac52-7fd7bfe2ee15)

**Comment by johnpaulcas** [comment]
Hi conneckit, just doing followup on this issue, thanks!

**Reply by lochie** [reply, marked as answer]
The problem here is the secondary wallet trying to override MetaMask connectivity which stops MetaMask being able to be used. This is something we're actively looking to resolve both [in ConnectKit](https://github.com/family/connectkit/pull/256) and via [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963).

PR #256 will hopefully resolve this for most wallets, but we can't guarantee _every_ wallet until IEP-6936 is in prod.

**Reply by johnpaulcas** [reply]
Hello @lochie , thank you for your prompt response. We are presently in the production phase, and I was wondering if there are any temporary solutions we can implement in the meantime? Also, could you possibly provide an estimated timeframe for when that PR is likely to be merged? Thank you.

**Reply by lochie** [reply]
Our PR to resolve some of these conflicts should be added in next week, but I'm unsure when EIP-6963 will be moved into production since that's a prop that is trying to be future-proofed with many [in-depth discussions](https://github.com/ethereum/EIPs/pull/7167) currently in the works.

Unfortunately this is one of the difficult things with creating a wallet connection flow where the wallets are all playing by their own rules, so we're constantly having to push patches to cover each wallets behaviour, so until IEP-6963 is in prod we're having to play keep-up.

**Reply by johnpaulcas** [reply]
hi @lochie want to follow up regards on the PR that possibly fixes this issue, thanks!
